src/snowflake/snowpark/mock/pandas_util.py

```diff
@@ -0,0 +1,214 @@
+                plain_data[row_idx][col_idx] = plain_data[row_idx][col_idx].value
+            elif isinstance(plain_data[row_idx][col_idx], pandas.Interval):
+
+                def convert_to_python_obj(obj):
+                    if isinstance(obj, numpy.float_):
+                        return float(obj)
+                    elif isinstance(obj, numpy.int_):
+                        return int(obj)
+                    elif isinstance(obj, pandas.Timestamp):
+                        return int(obj.value / 1000)
+                    else:
+                        raise NotImplementedError(
+                            f"[Local Testing] {type(obj)} within pandas.Interval is not supported."
+                        )
+
+                plain_data[row_idx][col_idx] = {
+                    "left": convert_to_python_obj(plain_data[row_idx][col_idx].left),
+                    "right": convert_to_python_obj(plain_data[row_idx][col_idx].right),
+                }
+            elif isinstance(plain_data[row_idx][col_idx], str):
+                pass
+            elif isinstance(plain_data[row_idx][col_idx], pd.Period):
+                # snowflake returns the ordinal of a period object
+                plain_data[row_idx][col_idx] = plain_data[row_idx][col_idx].ordinal
+            else:
+                previous_inferred_type = inferred_type_dict.get(col_idx)
+                data_type = infer_type(plain_data[row_idx][col_idx])
+                if isinstance(data_type, (MapType, ArrayType)):
+                    # snowflake converts python dict/array to variant
+                    data_type = VariantType()
+                if isinstance(data_type, DecimalType):
+                    # we need to calculate the precision and scale
+                    decimal_str = str(plain_data[row_idx][col_idx])
+                    decimal_parts = decimal_str.split(".")
+                    integer_len = (
+                        len(decimal_str)
+                        if len(decimal_parts) == 1
+                        else len(decimal_parts[0])
+                    )
+                    scale = 0 if len(decimal_parts) == 1 else len(decimal_parts[1])
+                    precision = integer_len + scale
+                    if precision > 38:
+                        raise SnowparkClientException(
+                            f"[Local Testing] Column precision {precision} and scale {scale} are not supported."
+                        )
+                    # handle integer and float separately
+                    data_type = DecimalType(precision=precision, scale=scale)
+                if previous_inferred_type:
+                    if isinstance(previous_inferred_type, NullType):
+                        inferred_type_dict[col_idx] = data_type
+                    if type(data_type) != type(previous_inferred_type):
+                        raise SnowparkClientException(
+                            "[Local Testing] Column contains different type of data."
```

Comment:
Let's list the two inferred types and change the error message to something like "Detected type x and type y in column, coercion is not currently supported..."

Reply:
updated